vcap/vcap/loading/capsule_loading.py

```diff
@@ -121,13 +132,36 @@ def load_capsule(path: Union[str, Path],
     try:
         _validate_capsule(new_capsule)
     except InvalidCapsuleError as e:
-        logging.warning(f"Failed to load capsule {path}")
         new_capsule.close()
         raise e
 
     return new_capsule
 
 
+def load_capsule_file(path: Union[str, Path],
+                      key: Optional[str] = None,
+                      inference_mode: bool = True) -> BaseCapsule:
+    """Load a capsule from the filesystem.
+
+    :param path: The path to the capsule file
+    :param key: The AES key to decrypt the capsule with, or None if the capsule
+        is not encrypted
+    :param inference_mode: If True, the backends for this capsule will be
+        started. If False, the capsule will never be able to run inference, but
+        it will still have it's various readable attributes.
+    """
+    path = Path(path)
+    source_path = (path.parent / path.stem).absolute()
```

Comment:
Do we want this assumption about where the source code would be inside the library? It might be a BrainFrame specific pattern.

Reply:
So here the assumption is that you have `/path/to/capsule.extension` and that the user would want to have tracebacks that read as though they lead to `/path/to/capsule`. I think that's a fine assumption, not one necessarily for BrainFrame

Reply:
Do you think this function should have an optional `source_path` field that, if not set, defaults to the current behavior?

Reply:
That's an interesting thought... So to confirm.

Some workflow a:
```
.
└── capsule-dev-dir
    ├── detector_person
    │   └── capsule.py
    └── detector_person.cap
```

Some workflow b:
```
.
├── capsule-dev-dir
│   └── detector_person.cap
└── some-source-code-storage-directory
    └── detector_person
        └── capsule.py
```

So in our current assumption, when you load a capsule and you have `workflow A`, everything will work fine. Basically, any tracebacks will lead to an actual, real path. Your changes would allow someone to potentially support `workflow B`, if they cared to pass the path to the source code. 

My thought initially is that it might be useful, as you said, especially if it defaults to the current behavior.

Reply:
Okay, added!